pom.xml

```diff
@@ -90,4 +90,16 @@
             </dependency>
         </dependencies>
     </dependencyManagement>
+
+    <profiles>
+        <profile>
+            <id>jdk17_plus</id>
+            <activation>
+                <jdk>[17,)</jdk>
+            </activation>
+            <modules>
+                <module>modern</module>
```

Comment:
I am not sure how many fancy new compiler features Jdbi will want to adopt, but it might be good to name the profiles after the JDK version they target. We might want a jdk17 module, and later want to add a jdk20 module, but the name `modern` can only be used once.

Reply:
ok, makes sense. Right now I am using this for JDK 17+ tests (such as records).

Reply:
Oh, if it will only be tests, it probably doesn't matter.

Reply:
right now there are only tests. It may eventually evolve into actual java 17 code. I renamed it "java17" and made it internal for now.